src/components/Avro/BaseAvroRecord.ts

```diff
@@ -1,26 +1,35 @@
 import * as avro from "avsc";
-import { Schema, Type } from "avsc";
-import { Memoize } from "typescript-memoize";
+import { Type } from "avsc";
+import * as mem from "mem";
 import { AvroRecord } from "./AvroRecord";
 
+function cacheKeyForSchema(schema: any): string {
+    return schema.namespace + schema.name;
+}
+
+const memoizedTypeForSchema = mem(
+    (schema: any): Type => avro.Type.forSchema(schema),
+    { cacheKey: cacheKeyForSchema },
+);
+
 export abstract class BaseAvroRecord implements AvroRecord {
 
     public static readonly subject: string = "";
     public static readonly schema = {};
 
-    @Memoize((schema: any) => {
-        return schema.namespace + schema.name;
-    })
     public static getTypeForSchema(schema: any): Type {
-        return avro.Type.forSchema(schema);
+        return memoizedTypeForSchema(schema);
     }
 
+    /*
+    I don't understand how this works, or where it's used
     @Memoize((schema: any) => {
```

Comment:
I believe this is some compiler magic going on here. Do you mean to comment out the static method as well for the resolver?